// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum AssuredworkloadsWorkloadResourcesResourceTypeEnum {
  AssuredworkloadsWorkloadResourcesResourceTypeEnumNO_VALUE_DO_NOT_USE = 0;
  AssuredworkloadsWorkloadResourcesResourceTypeEnumRESOURCE_TYPE_UNSPECIFIED = 1;
  AssuredworkloadsWorkloadResourcesResourceTypeEnumCONSUMER_PROJECT = 2;
  AssuredworkloadsWorkloadResourcesResourceTypeEnumENCRYPTION_KEYS_PROJECT = 3;
}

enum AssuredworkloadsWorkloadComplianceRegimeEnum {
  AssuredworkloadsWorkloadComplianceRegimeEnumNO_VALUE_DO_NOT_USE = 0;
  AssuredworkloadsWorkloadComplianceRegimeEnumCOMPLIANCE_REGIME_UNSPECIFIED = 1;
  AssuredworkloadsWorkloadComplianceRegimeEnumIL4 = 2;
  AssuredworkloadsWorkloadComplianceRegimeEnumCJIS = 3;
  AssuredworkloadsWorkloadComplianceRegimeEnumFEDRAMP_HIGH = 4;
  AssuredworkloadsWorkloadComplianceRegimeEnumFEDRAMP_MODERATE = 5;
  AssuredworkloadsWorkloadComplianceRegimeEnumUS_REGIONAL_ACCESS = 6;
}

message AssuredworkloadsWorkload {
  string name = 1;
  string display_name = 2;
  repeated AssuredworkloadsWorkloadResources resources = 3;
  AssuredworkloadsWorkloadComplianceRegimeEnum compliance_regime = 4;
  string create_time = 5;
  string billing_account = 6;
  map<string, string> labels = 7;
  string provisioned_resources_parent = 8;
  AssuredworkloadsWorkloadKmsSettings kms_settings = 9;
  string organization = 10;
  string location = 11;
}

message AssuredworkloadsWorkloadResources {
  int64 resource_id = 1;
  AssuredworkloadsWorkloadResourcesResourceTypeEnum resource_type = 2;
}

message AssuredworkloadsWorkloadKmsSettings {
  string next_rotation_time = 1;
  string rotation_period = 2;
}

message ApplyAssuredworkloadsWorkloadRequest {
  AssuredworkloadsWorkload resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteAssuredworkloadsWorkloadRequest {
  string service_account_file = 1;
  AssuredworkloadsWorkload resource = 2;
}

message ListAssuredworkloadsWorkloadRequest {
  string service_account_file = 1;
  string Organization = 2;
  string Location = 3;
}

message ListAssuredworkloadsWorkloadResponse {
  repeated AssuredworkloadsWorkload items = 1;
}

service AssuredworkloadsWorkloadService {
  rpc ApplyAssuredworkloadsWorkload(ApplyAssuredworkloadsWorkloadRequest) returns (AssuredworkloadsWorkload);
  rpc DeleteAssuredworkloadsWorkload(DeleteAssuredworkloadsWorkloadRequest) returns (google.protobuf.Empty);
  rpc ListAssuredworkloadsWorkload(ListAssuredworkloadsWorkloadRequest) returns (ListAssuredworkloadsWorkloadResponse);
}
